src/Acuminator/Acuminator.Analyzers/StaticAnalysis/RowChangesInEventHandlers/VariablesWalker.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Acuminator.Analyzers.StaticAnalysis.RowChangesInEventHandlers
+{
+	public partial class RowChangesInEventHandlersAnalyzer
+	{
+		/// <summary>
+		/// Collects all variables that are declared inside the method, and assigned with <code>e.Row</code>
+		/// </summary>
+		private class VariablesWalker : CSharpSyntaxWalker
+		{
+			private readonly SemanticModel _semanticModel;
+			private readonly PXContext _pxContext;
+			private CancellationToken _cancellationToken;
+			private readonly ImmutableHashSet<ILocalSymbol> _variables;
+
+			private readonly ISet<ILocalSymbol> _result = new HashSet<ILocalSymbol>();
+			public ImmutableArray<ILocalSymbol> Result => _result.ToImmutableArray();
+
+			public VariablesWalker(MethodDeclarationSyntax methodSyntax, SemanticModel semanticModel, PXContext pxContext,
+				CancellationToken cancellationToken)
+			{
+				methodSyntax.ThrowOnNull(nameof (methodSyntax));
+				semanticModel.ThrowOnNull(nameof (semanticModel));
+				pxContext.ThrowOnNull(nameof (pxContext));
+
+				_semanticModel = semanticModel;
+				_pxContext = pxContext;
+				_cancellationToken = cancellationToken;
+
+				if (methodSyntax.Body != null || methodSyntax.ExpressionBody?.Expression != null)
+				{
+					var dataFlow = methodSyntax.Body != null
+						? semanticModel.AnalyzeDataFlow(methodSyntax.Body)
+						: semanticModel.AnalyzeDataFlow(methodSyntax.ExpressionBody.Expression);
+
+					if (dataFlow.Succeeded)
+					{
+						_variables = dataFlow.WrittenInside
+							.Intersect(dataFlow.VariablesDeclared)
+							.OfType<ILocalSymbol>()
+							.ToImmutableHashSet();
+					}
+				}
+				
+			}
+
+			public override void VisitAssignmentExpression(AssignmentExpressionSyntax node)
+			{
+				_cancellationToken.ThrowIfCancellationRequested();
+
+				if (node.Left is IdentifierNameSyntax variableNode && node.Right != null)
```

Comment:
@dropsonic this will miss the following case:
```
MyDac firstVar, secondVar;
firstVar = secondVar = e.Row as MyDac;    //secondVar is missed

```

While it's not a frequent case still there are places in Pure where multi assignments are used.